enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,262 @@
+
+### Implementation Details/Notes/Constraints
+
+Operators supporting multiple architectures cannot stop supporting a given architecture since it would break heterogeneous clusters which are trying to upgrade.
+
+### Risks and Mitigations
+
+Releases and upgrade graphs for all architectures have to move simultaneously and cannot be independent. If there is a bug on one architecture, it will cause the edge or the release image to be removed for all architectures.
+
+## Design Details
+
+### RHCOS Support for ARM Azure
+
+The RHCOS pipeline will need to generate Azure disk images for ARM. However as of today, RHEL does not support hyper-v for ARM and that support is landing in RHEL8.6. Hence 4.11 builds of RHCOS for ARM Azure will be based on RHEL8.6. It is TBD whether a separate pipeline with early access RHCOS builds will be needed as RHEL8.6 would not have GA'ed before 4.11.
+
+### OCP Build & Release
+
+Phase 0 will involve building and publishing of [nightly builds][OCPPLAN-7640]. The release payload will be a manifest listed image with the component images being manifest listed as well. All 4 architectures - x86, ARM, Power and Z will be included in the manifestlist. Existing architecture specific release payloads continue to be delivered through their individual
+release controllers. A new heterogeneous release controller is bootstrapped and displays "4-stable" entries for ART constructed release payloads.
+
+`oc adm release new` would be [enhanced][WRKLDS-370] to allow the creation of manifestlist release payloads. The manifestlist flow would be triggered whenever the CVO image in an imagestream is a manifestlist. If the CVO image is a standard manifest, the generated release payload will also be a manifest. If the CVO image is a manifestlist, the generated release
+payload would be a manifestlist(containing a manifest for each architecture possessed by the CVO manifestlist). In either case, oc adm release new would permit non-CVO component images to be manifest or manifestlists and pass them through directly to the resultant release manifest(s). If a manifestlist release payload is generated, each architecture specific
+release payload manifest will reference the same pullspecs provided in the input imagestream.
+
+Since the internal Openshift registry does not support manifestlists, the release controller would use quay.io. There will be a new architecture option added to Cincinnati called "multi".
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the homogenous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for an additional graph for heterogeneous payloads which CVO will need to look at to make upgrade recommendations. There are a few upgrade scenarios to consider with the addition of heterogeneous payloads:
+
+- heterogeneous -> heterogeneous payload upgrade
+- homogenous -> heterogeneous payload upgrade
+- heterogeneous -> homogenous payload upgrade
```

Comment:
yeah , there is really no reason a user would need to go back to homogeneous except if they want to cut down on mirroring cost in the case of disconnected install when they decide to remove the node with the non control plane arch. I talked to Duncan and he felt the same so, I guess we can remove this.

Reply:
since the plan is to not support CI for heterogeneous, the only question is pre-flight upgrade. can we just document that when you upgrade back to homogeneous to make sure the nodepools are not heterogeneous?

Reply:
resolved after latest discussion.